src/intrinsics/mod.rs

```diff
@@ -464,33 +452,16 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     ),
                     _ => {}
                 }
+                // This cannot be a NaN so we also don't have to apply any non-determinism.
+                // (Also, `binary_op` already called `generate_nan` if needed.)
```

Comment:
This comment also got moved up incorrectly.

Please always make sure the comment logically talks about the point in the code where it sits. If the comment refers to something in past tense that will happen further down the code, that makes no sense.